File: algorithms/implementation/repeated-string/repeated-string-python/tests/test_repeatedString.py
from repeatedString import repeatedString

class TesteRepeatedString:
    
    def testeCaso1(self):
        
        s = 'aba'
        
        n =  10
        
        esperado = 7
        
        resultado = repeatedString(s, n)
        
        assert resultado == esperado
        
        
    def testeCaso1(self):
        
        s = 'a'
        
        n =  1000000000000
        
        esperado = 1000000000000
        
        resultado = repeatedString(s, n)
        
        assert resultado == esperado
        
        
    def testeCaso1(self):
        
        s = 'aab'
        
        n =  882787
        
        esperado = 588525
        
        resultado = repeatedString(s, n)
        
        assert resultado == esperado

File: algorithms/implementation/repeated-string/repeated-string-javascript/js/test_repeatedString.js
import {repeatedString} from '../js/repeatedString.js';
import assert from 'assert';

describe('repeatedString', function() {

    it('A frequência de "a" na substring é: 7', function() {

        let s = 'aba';
        
        let n =  10;
        
        let esperado = 7;
        
        let resultado = repeatedString(s, n);

        assert.strictEqual(resultado, esperado);
    });
});

describe('repeatedString', function() {

    it('A frequência de "a" na substring é: 1000000000000', function() {

        let s = 'a'
        
        let n =  1000000000000
        
        let esperado = 1000000000000
        
        let resultado = repeatedString(s, n);

        assert.strictEqual(resultado, esperado);
    });
});

describe('repeatedString', function() {

    it('A frequência de "a" na substring é: 588525', function() {

        let s = 'aab'
        
        let n =  882787
        
        let esperado = 588525
        
        let resultado = repeatedString(s, n);

        assert.strictEqual(resultado, esperado);
    });
});